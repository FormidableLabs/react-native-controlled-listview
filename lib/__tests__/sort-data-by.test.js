/*global it, expect*/
import 'react-native';
import sortDataBy from '../sort-data-by';
import people from './__data__/people';

it('sorts items when given a standard array comparator', () => {
  const comparator = (a, b) => {
    if (a.lastName > b.lastName) {
      return +1;
    }
    if (b.lastName < b.lastName) {
      return -1;
    }
    return 0;
  };

  expect(sortDataBy(comparator, people)).toMatchSnapshot();
});

it('sorts items when given a boolean array comparator', () => {
  const comparator = (a, b) => {
    return a.lastName < b.lastName;
  };

  expect(sortDataBy(comparator, people)).toMatchSnapshot();
});
