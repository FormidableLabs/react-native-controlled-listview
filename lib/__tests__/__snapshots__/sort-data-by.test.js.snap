exports[`test sorts items when given a boolean array comparator 1`] = `
Array [
  Object {
    "firstName": "Mildred",
    "lastName": "Bonk",
  },
  Object {
    "firstName": "Don",
    "lastName": "Gately",
  },
  Object {
    "firstName": "Hal",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Avril",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Orin",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Michael",
    "lastName": "Pemulis",
  },
]
`;

exports[`test sorts items when given a standard array comparator 1`] = `
Array [
  Object {
    "firstName": "Mildred",
    "lastName": "Bonk",
  },
  Object {
    "firstName": "Don",
    "lastName": "Gately",
  },
  Object {
    "firstName": "Orin",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Avril",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Hal",
    "lastName": "Incandenza",
  },
  Object {
    "firstName": "Michael",
    "lastName": "Pemulis",
  },
]
`;
